targets/TARGET_NUVOTON/TARGET_M480/crypto/crypto-misc.cpp

```diff
@@ -19,26 +19,46 @@
 #include "mbed_assert.h"
 #include "mbed_critical.h"
 #include "mbed_error.h"
+#include "cmsis_os2.h"
+#include "mbed_rtos_storage.h"
+#include <string.h>
 #include <limits.h>
 #include "nu_modutil.h"
 #include "nu_bitutil.h"
 #include "crypto-misc.h"
+#include "SingletonPtr.h"
+#include "Mutex.h"
+
+/* Consideration for choosing proper synchronization mechanism
+ *
+ * 1. Except for SHA AC which doesn't support context save & restore, we lock all crypto AC
+ *    for just their real operation rather than the whole lifetime of their crypto context.
+ *    For SHA AC, we provide SHA S/W fallback when SHA AC is not available. This policy can
+ *    avoid deadlock. We choose mutex to synchronize access to crypto non-SHA AC. Known
```

Comment:
No priority inversion any more

Reply:
@kjbracey-arm I fixed the comment in https://github.com/ARMmbed/mbed-os/pull/9081/commits/4d882aa7ce63867757241e241dd48c15e27f55b2. Because mutex lock scope is limited to a short sequence of operations, priority inversion issue gets less serious, right?

Reply:
There is no priority inversion at all any more - that's the point of using a mutex. The RTOS can and does have priority inheritance for mutexes: https://www.keil.com/pack/doc/CMSIS/RTOS2/html/group__CMSIS__RTOS__MutexMgmt.html

Reply:
@kjbracey-arm I got it. Thanks for your information.